packages/eslint-plugin/src/util/applyDefault.ts

```diff
@@ -0,0 +1,35 @@
+import { deepMerge, isObjectNotArray } from './deepMerge';
+
+/**
+ * Pure function - doesn't mutate either parameter!
+ * Uses the default options and overrides with the options provided by the user
+ * @param defaultOptions the defaults
+ * @param userOptions the user opts
+ * @returns the options with defaults
+ */
+export function applyDefault<TUser extends any[], TDefault extends TUser>(
+  defaultOptions: TDefault,
+  userOptions: TUser | null
+): TDefault {
+  // clone defaults
+  const options: TDefault = JSON.parse(JSON.stringify(defaultOptions));
+
+  // eslint-disable-next-line eqeqeq
```

Comment:
I don't think this be permitted for this particular rule. This is a standard case for it and we should either use this rule or we shouldn't.

I don't think there is any downside in being explicit with the `or undefined` check here